### PR TITLE
feat: Added network_interface and network_interface_ipconfigurations

### PR DESCRIPTION
**Pull request by YihengQi**

### CODE REVIEW

File: resources/services/network/interfaces.go [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+	}
+	return nil
+}
+func fetchNetworkInterfaceIPConfigurations(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	ni, ok := parent.Item.(network.Interface)
+	if !ok {
+		return fmt.Errorf("expected to have network.Interface but got %T", parent.Item)
+	}
+	res <- *ni.IPConfigurations
+	return nil
```
Comment (amanenk) [inline resources/services/network/interfaces.go:336]: `nil pointer exception` possibility, check other similar places
```suggestion
        if ni.IPConfigurations != nil
	    res <- *ni.IPConfigurations
	}
	return nil
```

**Review by amanenk — CHANGES_REQUESTED** [review]
Nice job. 
A few small touches required.

**Comment by YihengQi** [comment]
thanks for the feedback! 

Any command I could run with lint locally?

File: resources/services/network/interfaces.go [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+			{
+				Name:        "hosted_workloads",
+				Description: "List of references to linked BareMetal resources.",
+				Type:        schema.TypeStringArray,
+				Resolver:    schema.PathResolver("InterfacePropertiesFormat.HostedWorkloads"),
+			},
+			{
+				Name:        "mac_address",
+				Description: "The MAC address of the network interface.",
+				Type:        schema.TypeString,
```
Comment (roneli) [inline resources/services/network/interfaces.go:119]: nit: would suggest to maybe use schema.TypeMacAddr? you can use the schema.MacAddressResolver

File: resources/services/network/interfaces.go [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+					{
+						Name:        "load_balancer_inbound_nat_rules",
+						Description: "A list of references of LoadBalancerInboundNatRules.",
+						Type:        schema.TypeJSON,
+						Resolver:    schema.PathResolver("InterfaceIPConfigurationPropertiesFormat.LoadBalancerInboundNatRules"),
+					},
+					{
+						Name:        "private_ip_address",
+						Description: "Private IP address of the IP configuration.",
+						Type:        schema.TypeString,
```
Comment (roneli) [inline resources/services/network/interfaces.go:261]: nit: would suggest to maybe use schema.TypeInet or schema.TypeIP

File: resources/services/network/interfaces.go [hunk 4]
```diff
@@ -0,0 +1,407 @@
...
+					{
+						Name:        "provisioning_state",
+						Description: "The provisioning state of the network interface IP configuration.",
+						Type:        schema.TypeString,
+						Resolver:    schema.PathResolver("InterfaceIPConfigurationPropertiesFormat.ProvisioningState"),
+					},
+					{
+						Name:        "public_ip_address",
+						Description: "Public IP address bound to the IP configuration.",
+						Type:        schema.TypeString,
```
Comment (roneli) [inline resources/services/network/interfaces.go:291]: nit: would suggest to maybe use schema.TypeInet or schema.TypeIP

**Review by roneli — APPROVED** [review]
LGTM, had a couple of suggestions on a few types.

File: resources/services/network/interfaces.go (hunk 2 above)
Comment (YihengQi) [reply resources/services/network/interfaces.go:119]: might just keep them like this for now and see how things go with iterations. Run into some problems with database types after changing.

**Pull request merged**